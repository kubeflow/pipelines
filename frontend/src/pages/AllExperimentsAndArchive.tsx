/*
 * Copyright 2018 The Kubeflow Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from 'react';
import ExperimentList from './ExperimentList';
import ArchivedExperiments from './ArchivedExperiments';
import MD2Tabs from '../atoms/MD2Tabs';
import { Page, PageProps } from './Page';
import { RoutePage } from '../components/Router';
import { ToolbarProps } from '../components/Toolbar';
import { classes } from 'typestyle';
import { commonCss, padding } from '../Css';

export enum AllExperimentsAndArchiveTab {
  EXPERIMENTS = 0,
  ARCHIVE = 1,
}

export interface AllExperimentsAndArchiveProps extends PageProps {
  view: AllExperimentsAndArchiveTab;
}

interface AllExperimentsAndArchiveState {
  selectedTab: AllExperimentsAndArchiveTab;
}

class AllExperimentsAndArchive extends Page<
  AllExperimentsAndArchiveProps,
  AllExperimentsAndArchiveState
> {
  public getInitialToolbarState(): ToolbarProps {
    return { actions: {}, breadcrumbs: [], pageTitle: '' };
  }

  public render(): JSX.Element {
    return (
      <div className={classes(commonCss.page, padding(20, 't'))}>
        <MD2Tabs
          tabs={['Active', 'Archived']}
          selectedTab={this.props.view}
          onSwitch={this._tabSwitched.bind(this)}
        />
        {this.props.view === 0 && <ExperimentList {...this.props} />}

        {this.props.view === 1 && <ArchivedExperiments {...this.props} />}
      </div>
    );
  }

  public async refresh(): Promise<void> {
    return;
  }

  private _tabSwitched(newTab: AllExperimentsAndArchiveTab): void {
    this.props.history.push(
      newTab === AllExperimentsAndArchiveTab.EXPERIMENTS
        ? RoutePage.EXPERIMENTS
        : RoutePage.ARCHIVED_EXPERIMENTS,
    );
  }
}

export default AllExperimentsAndArchive;
