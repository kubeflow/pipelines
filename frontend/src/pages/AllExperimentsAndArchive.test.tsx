/*
 * Copyright 2018 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from 'react';
import AllExperimentsAndArchive, {
  AllExperimentsAndArchiveProps,
  AllExperimentsAndArchiveTab,
} from './AllExperimentsAndArchive';
import { shallow } from 'enzyme';

function generateProps(): AllExperimentsAndArchiveProps {
  return {
    history: {} as any,
    location: '' as any,
    match: '' as any,
    toolbarProps: {} as any,
    updateBanner: () => null,
    updateDialog: jest.fn(),
    updateSnackbar: jest.fn(),
    updateToolbar: () => null,
    view: AllExperimentsAndArchiveTab.EXPERIMENTS,
  };
}

describe('ExperimentsAndArchive', () => {
  it('renders experiments page', () => {
    expect(shallow(<AllExperimentsAndArchive {...(generateProps() as any)} />)).toMatchSnapshot();
  });

  it('renders archive page', () => {
    const props = generateProps();
    props.view = AllExperimentsAndArchiveTab.ARCHIVE;
    expect(shallow(<AllExperimentsAndArchive {...(props as any)} />)).toMatchSnapshot();
  });

  it('switches to clicked page by pushing to history', () => {
    const spy = jest.fn();
    const props = generateProps();
    props.history.push = spy;
    const tree = shallow(<AllExperimentsAndArchive {...(props as any)} />);

    tree.find('MD2Tabs').simulate('switch', 1);
    expect(spy).toHaveBeenCalledWith('/archive/experiments');

    tree.find('MD2Tabs').simulate('switch', 0);
    expect(spy).toHaveBeenCalledWith('/experiments');
  });
});
