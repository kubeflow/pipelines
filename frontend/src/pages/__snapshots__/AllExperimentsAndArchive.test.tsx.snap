// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperimentsAndArchive renders archive page 1`] = `
<div
  className="page"
>
  <MD2Tabs
    onSwitch={[Function]}
    selectedTab={1}
    tabs={
      Array [
        "Active",
        "Archived",
      ]
    }
  />
  <EnhancedArchivedExperiments
    history={Object {}}
    location=""
    match=""
    toolbarProps={Object {}}
    updateBanner={[Function]}
    updateDialog={[MockFunction]}
    updateSnackbar={[MockFunction]}
    updateToolbar={[Function]}
    view={1}
  />
</div>
`;

exports[`ExperimentsAndArchive renders experiments page 1`] = `
<div
  className="page"
>
  <MD2Tabs
    onSwitch={[Function]}
    selectedTab={0}
    tabs={
      Array [
        "Active",
        "Archived",
      ]
    }
  />
  <EnhancedExperimentList
    history={Object {}}
    location=""
    match=""
    toolbarProps={Object {}}
    updateBanner={[Function]}
    updateDialog={[MockFunction]}
    updateSnackbar={[MockFunction]}
    updateToolbar={[Function]}
    view={0}
  />
</div>
`;
